source/winVersion.py

```diff
@@ -49,12 +50,23 @@ def _windowsVersionToReleaseName(self):
 		elif (self.major, self.minor) == (6, 3):
 			return "Windows 8.1"
 		elif self.major == 10:
-			buildsToReleases = {build: release for release, build in WIN10_RELEASE_NAME_TO_BUILDS.items()}
-			if self.build in buildsToReleases:
-				return f"Windows 10 {buildsToReleases[self.build]}"
-			else:
-				# Windows Insider build.
-				return "Windows 10 prerelease"
+			# From Version 1511 (build 10586), release Id/display version comes from Windows Registry.
+			# Always return "Windows 10 1507" on build 10240.
+			if self.build == 10240:
+				return "Windows 10 1507"
+			with winreg.OpenKey(
+				winreg.HKEY_LOCAL_MACHINE, r"Software\Microsoft\Windows NT\CurrentVersion"
+			) as currentVersion:
+				# Version 20H2 and later where a separate display version string is used.
+				# For backward compatibility, release Id will store display version string.
+				try:
+					releaseId = winreg.QueryValueEx(currentVersion, "DisplayVersion")[0]
+				except OSError:
+					releaseId = None
+				# Version 1511 and later unless display version string is present.
+				if not releaseId:
+					releaseId = winreg.QueryValueEx(currentVersion, "ReleaseID")[0]
+			return f"Windows 10 {releaseId}"

```

Comment:
This will return the name of the Windows release you are running on regardless of how the `WinVersion` class is constructed right?

Eg, ensure there is a unit test for this. I suspect this will fail on any computer running a later version of Windows.
```python
assertEqual(repr(WIN10_1511), "Windows 10 10586")
```

Reply:
If you keep this for the `getWinVer` case, please cache it at module level, it doesn't need to happen more than once.